v5.2.0 design refresh, plus responsive offcanvas classes

mdo opened the pull request:
![bootstrap-v520-home](https://user-images.githubusercontent.com/98681/151459990-1e15aae2-6855-4c98-a299-dd2028b4a79e.png)

![bootstrap-v520-docs](https://user-images.githubusercontent.com/98681/151460566-bd823101-c030-46c5-8895-d25a56f9dac2.png)

This is a redesign of the homepage and docs layout to give v5.2.0 some new life. I've been playing around with this while recovering from my heart attack, so not everything is perfect, but it feels refreshing to me. And while the design has been updated, so too has the homepage content.

Here are the key changes:

- **New homepage design and content.** I've rewritten the homepage to better articulate what comes with Bootstrap and how it's powerful/useful to folks. Key content additions are the utility API, JavaScript plugins, and increasing CSS variable usage.

- **New documentation sidebar.** The sidebar of the docs pages has been updated to include colorful Bootstrap Icons as part of the section headers and an always expanded list of links. I've also written some JS (lol) to automatically scroll the sidebar to the `.active` link, so you can always tell where you are and quickly jump around that section.

- **Cross-linking page versions is coming.** I've added a first step at cross-linking the same page across the various versions, unless the page was added in another release (e.g., with the CSS Grid page in v5.1.0). Now in the dropdown, we'll give you a link back to each release and show a disabled version number if the page isn't a part of that previous release.

- **Adds a handful of new utilities.** Bootstrap now includes `.fw-semibold`, which we use across our new homepage, and additional `.border-*` sizes with `.border-4` and `.border-5`.

- **Refreshes `border-radius` on buttons and form controls.** You might notice that buttons, inputs, and more have been slightly tweaked with increased radii on their corners. This is purely an aesthetic change to make buttons look a little more button-y (and our form controls updated to match).

- **Adds responsive offcanvas support outside navbars.** Redesigns the navbar in our docs to use this to collapse our sidebar on narrow viewports. Fixes #33387.

There are some other miscellaneous updates to fine-tune things as well. I've fixed some inconsistent padding between docs navbar, subnav, content, and footer. I've also tweaked some page copy here and there as part of the refresh.

Fixes #33993 as well.

JS changes from @GeoSot fixes #35594.

---

### Todos

- [x] Consider reverting sidebar width change
- [x] Update the versions dropdown to include v5.1.3 links
- [ ] Consider cutting a v5.1.4 release with docs version picker changes? Or just push to `gh-pages`? _This would cross-link v5.1.x pages with their v5.0.x counterparts._
- [x] Double check any optimizations we might need (e.g., newly added custom icons sprite)
- [x] Document new `border` and `font-weight` utilities
- [x] Sidebar is too tall on mobile hurting usability since the user needs to scroll a lot
- [x] Work on JS and fix tests

Preview: https://deploy-preview-35736--twbs-bootstrap.netlify.app/

patrickhlauke commented:
otherwise though, looking nice a modern and fresh. good stuff! will dig in a bit deeper later

XhmikosR commented on site/assets/scss/_content.scss:
```diff
@@ -79,3 +79,35 @@
+  align-items: center;
+  text-decoration-color: rgba($primary, .5);
+  text-underline-offset: .5rem;
+  -webkit-backface-visibility: hidden; // stylelint-disable-line property-no-vendor-prefix
```
Maybe we should use the unprefixed property and let autoprefixer handle these?

alecpl commented:
I don't like the radius change. `0.4rem` radius border looks "distorted" (more in Firefox, for some reason) for me.

mdo commented:
> super nitpicky: with the page background now `#FEFFFF` rather than `#FFFFFF`, the blue links, pink/red code, in the prose now fall just below 4.5:1 contrast. it may be best to just bump the page background back to full white, rather than going with the "you failed by just a smidge" on the contrast.

The `body` `background-color` is still `#fff`.

> I don't like the radius change. `0.4rem` radius border looks "distorted" (more in Firefox, for some reason) for me.

Not seeing any distortion here.

patrickhlauke commented:
> The `body` `background-color` is still `#fff`.

Oops, my fault, was going just by the screenshot rather than firing it up live (and oddly, the .png has a mix of `#ffffff`, `#efffff`, `#feffff` etc in the background, some compression artifact)

![image](https://user-images.githubusercontent.com/895831/151655525-88c4293a-0676-4501-9ff3-297840c6f577.png)

![image](https://user-images.githubusercontent.com/895831/151655530-b160f3a4-1f00-4d34-a09d-82282b9e5312.png)

Anyway, ignore that then :)

XhmikosR commented on site/content/docs/5.1/components/placeholders.md:
```diff
@@ -4,6 +4,7 @@ title: Placeholders
 description: Use loading placeholders for your components or pages to indicate something may still be loading.
 group: components
 toc: true
+added: "5.1"
```
Are these used?

mdo replied:
This frontmatter tells the version picker in the subnav that if you're browsing v5.1.x and you want to see the placeholders page in the previous version from that dropdown (v5.0.2), you cannot, because the page wasn't added until v5.1.

XhmikosR commented on site/data/plugins.yml:
```diff
@@ -0,0 +1,47 @@
+- name: Alert
```
Couldn't we get this info via a front matter property and the page description? Just wondering if we really need a new file, not against it per se :)

mdo replied:
Open to that as an optimization in the future. For now, keeping this as-is.

XhmikosR commented on site/layouts/_default/baseof.html:
```diff
@@ -5,6 +5,7 @@
   </head>
   {{ block "body_override" . }}<body>{{ end }}
     {{ partial "skippy" . }}
+    {{ partial "icons" . }}
```
Does the sprite include only the needed icons? If so, would be nice if we had a quick command documented to generate them in the future.

mdo replied:
I manually created this, copying and pasting and editing from the Bootstrap Icons. It does only include icons that are needed across the homepage and docs layouts. Not much is wasted between the two right now, and I'll likely add even more icons to the homepage in another commit.

XhmikosR commented on site/layouts/partials/favicons.html:
```diff
@@ -3,6 +3,6 @@
 <link rel="icon" href="/docs/{{ .Site.Params.docs_version }}/assets/img/favicons/favicon-16x16.png" sizes="16x16" type="image/png">
 <link rel="manifest" href="/docs/{{ .Site.Params.docs_version }}/assets/img/favicons/manifest.json">
-<link rel="mask-icon" href="/docs/{{ .Site.Params.docs_version }}/assets/img/favicons/safari-pinned-tab.svg" color="#7952b3">
+<link rel="mask-icon" href="/docs/{{ .Site.Params.docs_version }}/assets/img/favicons/safari-pinned-tab.svg" color="#712cf9">
```
Should we generate new favicons/social images too?

XhmikosR commented on scss/_utilities.scss:
```diff
@@ -461,6 +461,7 @@ $utilities: map-merge(
         lighter: $font-weight-lighter,
         normal: $font-weight-normal,
         bold: $font-weight-bold,
+        semibold: $font-weight-semibold,
```
How about moving the new utilities to a separate PR and then rebase this one?

mdo replied:
Rebased everything to isolate these changes to their own commit, I think that's okay since it's clearly outlined.

GeoSot commented on scss/_variables.scss:
```diff
@@ -488,9 +488,11 @@ $border-color:                $gray-300 !default;
-$border-radius:               .25rem !default;
-$border-radius-sm:            .2rem !default;
-$border-radius-lg:            .3rem !default;
+$border-radius:               .4rem !default;
```
I am not sure, but this is the base variables file. Right? 
If we change the variables here, wouldn't be a breaking change, for dist files and for all that haven't override the radius variables?

mdo replied:
I'm changing the default values, but I don't see this as a breaking change. Going for a visual refresh on a few components alongside our docs.

GeoSot commented on site/assets/js/application.js:
```diff
@@ -119,7 +134,9 @@
   var btnTitle = 'Copy to clipboard'
-  var btnHtml = '<div class="bd-clipboard"><button type="button" class="btn-clipboard">Copy</button></div>'
+  // var btnHtml = '<div class="bd-clipboard"><button type="button" class="btn-clipboard">Copy</button></div>'
+  var btnHtml = '<div class="bd-clipboard"><button type="button" class="btn-clipboard" title="Copy to clipboard"><svg class="bi" width="1em" height="1em" fill="currentColor"><use xlink:href="#clipboard"/></svg></button></div>'
```
```suggestion
  var btnHtml = '<div class="bd-clipboard"><button type="button" class="btn-clipboard" ><svg class="bi" width="1em" height="1em" fill="currentColor"><use xlink:href="#clipboard"/></svg></button></div>'
```
we can omit title as it is initialized on 147 line

GeoSot commented on site/assets/js/application.js:
```diff
@@ -144,13 +161,23 @@
+    iconFirstChild.setAttributeNS(namespace, 'href', originalXhref.replace('clipboard', 'check2'))
+
+    setTimeout(function () {
+      iconFirstChild.setAttributeNS(namespace, 'href', originalXhref)
```
`setTimeout` maybe could be avoided, if we put the code-block on line 172 (inside hidden tooltip event)

mdo replied:
Thanks, just tried that. While it works, I prefer the not-so-instant change of the icon and tooltip. It feels a little jarring without the timeout, so I'm going to leave it as-is for now :).

ffoodd commented on site/assets/js/application.js:
```diff
@@ -119,7 +134,9 @@
   var btnTitle = 'Copy to clipboard'
-  var btnHtml = '<div class="bd-clipboard"><button type="button" class="btn-clipboard">Copy</button></div>'
+  // var btnHtml = '<div class="bd-clipboard"><button type="button" class="btn-clipboard">Copy</button></div>'
+  var btnHtml = '<div class="bd-clipboard"><button type="button" class="btn-clipboard" ><svg class="bi" width="1em" height="1em" fill="currentColor"><use xlink:href="#clipboard"/></svg></button></div>'
```
This needs some accessibility love: I'd go with `role="img" aria-label="Copy"` on the SVG, just in case tooltip doesn't work or something.

And feel free to drop the previous line instead of commenting, I thought it was some docs :D

ffoodd commented on site/assets/scss/_component-examples.scss:
```diff
@@ -330,6 +330,11 @@
+
+    // Undo tabindex that's automatically added by Hugo
+    &:focus {
+      outline: 0;
```
Consider using `outline-color: transparent` instead: that'd allow OS' high constrasts modes to better operate.

mdo replied:
We use `outline: 0` in around a dozen other spots, maybe we need a more holistic look at that? Going to keep as-is because of that for now.

ffoodd replied:
Seems legit, we'll address that everywhere at once (probably after your global focus vars PR).

ffoodd commented on site/assets/scss/_masthead.scss:
```diff
@@ -7,32 +7,70 @@
+    top: .5rem;
+  }
+
+  #carbonads { // stylelint-disable-line selector-max-id
+    margin-right: auto;
+    margin-left: auto;
+  }
```
Could we imagine using utilities for such things? `.mx-auto` should be fine, isn't it? And there're quite a few declarations in site styles that we may drop. Could be tackled later, obviously.

mdo replied:
All the HTML is added dynamically, so I cannot modify it with utilities.

ffoodd commented on site/assets/scss/_toc.scss:
```diff
@@ -40,3 +40,42 @@
+    }
+  }
+
+  @include media-breakpoint-up(md) {
+    pointer-events: none;
+  }
```
I think the button should be hidden on large screens, and replaced by a heading or something: currently, when navigating with keyboard, I'm still able to activate the button, which makes the ToC flicks visually (and makes `aria-*` irrelevant).

ffoodd commented on site/layouts/_default/docs.html:
```diff
@@ -18,12 +18,18 @@ <h1 class="bd-title" id="content">{{ .Title | markdownify }}</h1>
         <div class="bd-toc mt-4 mb-5 my-md-0 ps-xl-3 mb-lg-5 text-muted">
-          <strong class="d-block h6 my-2 pb-2 border-bottom">On this page</strong>
-          {{ .TableOfContents }}
+          <button class="btn btn-link link-dark p-md-0 mb-2 mb-md-0 fw-semibold text-decoration-none bd-toc-toggle" type="button" data-bs-toggle="collapse" data-bs-target="#tocContents" aria-expanded="false" aria-controls="tocContents">
```
Here it is: I'd go with `.d-lg-none` on the button itself, and another element (`p` or `hn` most likely) with `.d-none.d-lg-block`. WDYT?

ffoodd commented on site/layouts/_default/docs.html:
```diff
@@ -18,12 +18,18 @@ <h1 class="bd-title" id="content">{{ .Title | markdownify }}</h1>
-          {{ .TableOfContents }}
+          <button class="btn btn-link link-dark p-md-0 mb-2 mb-md-0 fw-semibold text-decoration-none bd-toc-toggle" type="button" data-bs-toggle="collapse" data-bs-target="#tocContents" aria-expanded="false" aria-controls="tocContents">
+            On this page
+            <svg class="bi d-md-none ms-2"><use xlink:href="#chevron-expand"></use></svg>
```
Needs some accessibility love too: `aria-hidden="true"` on `svg` should be good.

ffoodd commented on site/layouts/partials/docs-sidebar.html:
```diff
@@ -16,26 +16,27 @@
-      <li class="mb-1">
-        <button class="btn d-inline-flex align-items-center rounded{{ if not $is_active_group }} collapsed{{ end }}" data-bs-toggle="collapse" data-bs-target="#{{ $group_slug }}-collapse" aria-expanded="{{ $is_active_group }}"{{ if $is_active_group }} aria-current="true"{{ end }}>
+      <li class="bd-links-group py-2">
+        <h6 class="bd-links-heading d-flex w-100 align-items-center">
```
`h6` feels weird here, the headings hierarchy is quite messy: 
![image](https://user-images.githubusercontent.com/2890570/154106707-e4bf4242-970d-4e8c-8efa-bc12441d94ff.png)
Maybe use a `<p class="h6">` instead?
@patrickhlauke Is it important or not that much? I remember your "When headings fails WCAG" post.

ffoodd commented on site/layouts/partials/docs-sidebar.html:
```diff
@@ -16,26 +16,27 @@
+      <li class="bd-links-group py-2">
+        <h6 class="bd-links-heading d-flex w-100 align-items-center">
+          {{- if $group.icon }}
+            <svg class="bi me-2"{{- if $group.icon_color }} style="color: var(--bs-{{ $group.icon_color }});"{{- end }}><use xlink:href="#{{ $group.icon }}"></use></svg>
```
Since it's decorative, it needs some `aria-hidden="true"` too :)

ffoodd reviewed (commented):
Looks great, it's really refreshing to get some colors!

I noticed a few moves, mostly accessibility improvements.

As @alecpl, I'm seeing weird glitches with the new radii, here's what it looks like on my Firefox: 
![image](https://user-images.githubusercontent.com/2890570/154107520-6e0f2b8f-8dec-4fd9-9b39-70eb15d4332e.png)
I'm not sure what's happening here, maybe some subpixels rounding bug?

And regarding homepage, the first masthead get centered ("Get started anyway you want") which feels a bit strange since next blocks are left-aligned. I suppose it's to make things aligned with the main masthead?

The overall look & feel is very nice, I love it :)

rabuckley commented on site/layouts/partials/home/masthead-followup.html:
```diff
@@ -1,37 +1,39 @@
+      <svg class="bi fs-1"><use xlink:href="#code"></use></svg>
     </div>
-  </section>
+    <h2 class="display-5 mb-3 fw-semibold lh-sm">Get started anyway you&nbsp;want</h2>
```
```suggestion
    <h2 class="display-5 mb-3 fw-semibold lh-sm">Get started any way you&nbsp;want</h2>
```
any way (in any of the ways) vs anyway (regardless/nonetheless)

mdo commented:
Thanks @ffoodd, changes pushed. I adjusted the default `border-radius` value from `.4rem` which calculates to `6.4px` to `.375rem` which comes out to an even 6px. That should help with those corners looking off.

mdo commented:
Pushed some new changes to refine our navbar and its mobile behaviors.

- Navbar now has a slightly translucent gradient, making for a really cool effect IMO :).
- The search form and version picker has been moved to the main navbar, so we have no more need for the subnav which has now been removed. I'm wondering if we need to do something to upgrade to that new fangled Algolia search?
- I've also removed the Home link in the navbar because it's quite redundant next to the logo which links there.
- Lastly, I've moved the sidebar toggle to the navbar and switched up the icons for mobile views. TBD if I leave the `Browse` or similar word there for the docs sidebar menu toggle.

Have a look, let me know what you think—I'm really liking it.

![Screen Shot 2022-02-20 at 10 04 20 AM](https://user-images.githubusercontent.com/98681/154857335-27b37640-5b91-4819-8ca4-8f5d3b3aa856.png)

![Screen Shot 2022-02-20 at 10 04 27 AM](https://user-images.githubusercontent.com/98681/154857339-0403cd50-d9ad-4db1-a5e6-14b4508702cb.png)

mdo commented:
### New design update

Aww yeah, I've got another huge update. This time I've added responsive offcanvas classes (modifying the navbar offcanvas styles in the process) and converted our sidebar into it's own offcanvas in the process.

![new-docs-offcanvas-sidebar](https://user-images.githubusercontent.com/98681/155260545-d3f2ce94-2ba0-4146-83ca-51246b42c24e.png)

### Responsive offcanvas

New responsive classes for the offcanvas are available with a new demo added into the docs.

![responsive-offcanvas](https://user-images.githubusercontent.com/98681/155260829-9d91fee5-31a3-404a-a0cd-dbccf5065d07.png)

![responsive-offcanvas-open](https://user-images.githubusercontent.com/98681/155260833-f255474a-ab6e-410c-b5b7-2db2a693dd99.png)

### New Examples layout

I've also updated the Examples page to have a new layout.

![new-docs-examples](https://user-images.githubusercontent.com/98681/155260553-20db7ea7-fae8-4f11-b270-371841e58898.png)

ffoodd commented on site/layouts/_default/single.html:
```diff
@@ -1,14 +1,19 @@
-            <a href="{{ .Site.Params.download.source }}" class="btn btn-lg btn-outline-secondary mt-3 mt-sm-0 ms-sm-3" onclick="ga('send', 'event', 'Examples', 'Hero', 'Download');">Download source code</a>
+          <div class="d-flex flex-column flex-md-row gap-3">
+            <a href="{{ .Site.Params.download.dist_examples }}" class="btn btn-lg bd-btn-lg btn-bd-primary d-flex align-items-center justify-content-center fw-semibold" onclick="ga('send', 'event', 'Examples', 'Hero', 'Download Examples');">
+              <svg class="bi me-2"><use xlink:href="#box-seam"></use></svg>
```
Missing `aria-hidden="true"` here.
```suggestion
              <svg class="bi me-2" aria-hidden="true"><use xlink:href="#box-seam"></use></svg>
```

ffoodd commented on site/layouts/partials/docs-navbar.html:
```diff
@@ -1,74 +1,91 @@
-    </button>
+    <div class="d-flex d-lg-none order-3">
+      <button class="navbar-toggler p-2" type="button" data-bs-toggle="offcanvas" data-bs-target="#bdNavbar" aria-controls="bdNavbar" aria-expanded="false" aria-label="Toggle navigation">
+        <svg class="bi" width="24" height="24"><use xlink:href="#three-dots"></use></svg>
```
```suggestion
        <svg class="bi" width="24" height="24" aria-hidden="true"><use xlink:href="#three-dots"></use></svg>
```

ffoodd commented on site/layouts/partials/home/masthead.html:
```diff
@@ -1,28 +1,33 @@
-          <a href="/docs/versions/" class="link-secondary">All releases</a>
-        </p>
+        <a href="/docs/{{ .Site.Params.docs_version }}/getting-started/introduction/" class="btn btn-lg bd-btn-lg btn-bd-primary d-flex align-items-center justify-content-center fw-semibold" onclick="ga('send', 'event', 'Jumbotron actions', 'Get started', 'Get started');">
+          <svg class="bi me-2"><use xlink:href="#book-half"></use></svg>
```
```suggestion
          <svg class="bi me-2" aria-hidden="true"><use xlink:href="#book-half"></use></svg>
```

ffoodd reviewed (commented):
I really like it! It's so refreshing.

Regarding offcanvas, I think we need some JS love for responsive variants: opening an offcanvas and resizing viewport until it's not applicable anymore successfully hide the content, but the backdrop remains.

Not quite sure we really need to handle resizing, but that's the only concern I have right now :)

May I suggest we also kind of update cheatsheets at the same time? Latest changes (like responsive offcanvas) should appear in here too, I guess; and the new header could fit really well with the current cheatsheet page, refreshing it too. Current design makes obvious I'm not a designer I guess ;)

The upcoming 5.2 will be awesome, thanks for the inspiration :heart:

GeoSot commented:
>Regarding offcanvas, I think we need some JS love for responsive variants: opening an offcanvas and resizing viewport until it's not applicable anymore successfully hide the content, but the backdrop remains.

Did a small trick. If we are ok with this I can then add tests

mdo committed:
Add responsive offcanvas classes

- Updates navbar-expand classes to de-dupe some styles—these shouldn't interfere now.
- Adds some JS to the offcanvas component to help with responsiveness

Co-Authored-By: GeoSot <geo.sotis@gmail.com>

mdo committed:
Redesign homepage, docs, and examples

Homepage:

- New Bootstrap purple navbar
- Redesigned masthead
- Rewrote and redesigned homepage content
- Replace Copy text with icons like Bootstrap Icons site across all ClipboardJS instances
- Fixed padding issues in site footer
- Match homepage button styles to examples page, use gap instead of tons of responsive margin utils

Docs:

- New navbar, no more subnav. Migrated search and version picker into the main navbar and refreshed the design of it all, including the responsive toggles.
- New sidebar navigation is always expanded, and now features Bootstrap Icons alongside section headings
- Sidebar navigation autoscrolls to active link for better usability
- Subnav and navbar padding issues ironed out
- Enhanced the version picker in anticipation of v5.2: we can now link right to the same page in the previous version.
- Redesign callouts to add more color to our pages
- Collapse table of contents on mobile
- Cleanup and redesign button styles with CSS variables
- Update design for subnav version dropdown
- Update highlight and example to be full-width until md
- Improve the Added In badges
- Turn the ToC into a well on mobile
- Redesign code snippets to better house two action buttons

Examples:

- Redesign Examples page layout
- Add new example for responsive offcanvases in navbars

mdo committed:
Convert offcanvas to CSS vars

mdo commented:
@GeoSot I may have force-pushed over your commits at some point, I'm not sure. Any chance you can have another look?

GeoSot commented:
@mdo can we keep the base class `offcanvas` in responsive offcanvas`es too? 
We may face some issues with js selectors

Example: `offcanvas offcanvas-lg`

XhmikosR commented:
Just a quick comment, this PR introduces a lot of accessibility issues (contrast ratio, buttons etc).

We better fix the issues before landing it. Also, I still think the docs redesign should be untied from any utils addition since we've seen what this can cause from past PRs 😛

mdo commented:
Cherry-picked out a few things into separate PRs for y'all to review and then I'll redo this PR to remove those pieces and give it a big update.

- #35979
- #35980
- #35981

Please review when you can :).

patrickhlauke commented:
i've dropped the ball on reviewing this in much depth. let me know once it's stable enough, if there are still things getting reorganised/changed dramatically at this point. @XhmikosR could you share a list of concerns of what you've seen accessibility-wise, so I can make sure to check over those too?

on a completely silly point...i'm on a windows machine, so the mac command key ⌘ shortcut thing in the docs search seems a bit out of place...

![docs search box, showing the ⌘K shortcut](https://user-images.githubusercontent.com/895831/157658818-9b8a17fc-afcf-4de9-9c8b-5b8a9a1ec36f.png)

XhmikosR commented:
I'm using the bookmarklet from https://squizlabs.github.io/HTML_CodeSniffer/ and I see plenty of issues. This PR is now split into parts so some of the color issues can be seen in https://github.com/twbs/bootstrap/pull/35984

patrickhlauke commented:
> I'm using the bookmarklet from https://squizlabs.github.io/HTML_CodeSniffer/ and I see plenty of issues. This PR is now split into parts so some of the color issues can be seen in #35984

took it for a quick spin. most of the warnings seem to be ignorable (where they relate to transparent background colours, where the bookmarklet is unable to do an automated contrast calculation - will need manual checking). the hard error about lack of submit button in the form is debatable/wrong i'd say. but yes, will need to do a proper concerted run-through of the changes (and also check contrast for all colour combos in dark mode)

mdo commented:
The diff here has been dramatically updated with the removal of those other PRs mentioned in https://github.com/twbs/bootstrap/pull/35736#issuecomment-1063171257. Might be more I can pull out as I try to revisit the offcanvas changes that @GeoSot has suggested to me over Slack.

mdo commented:
@patrickhlauke Shortcut for the search bar should be resolved now. Just tested it myself.

patrickhlauke commented:
perhaps tweaking it even further to use `Ctrl+K` rather than `^K`? I'll admit that even I, as not exactly a noob, was initially a bit confused what the `^` meant

GeoSot committed:
Feat: add resize handler to Offcanvas.js.

If we could use as default the `.offcanvas` class without modifiers, we then, could add a simplified selector
The selector itself, ignores the .offcanvas class as it doesn't have any responsive behavior
The `aria-modal` addon is to protect us, selection backdrop elements

mdo commented:
Just pushed some more changes, including bringing in DocSearch v3. Still some more to clean up—including an odd jumping when you hit `Esc` after the DocSearch dialog pops up.

<img width="1792" alt="Screen Shot 2022-04-13 at 5 31 32 PM" src="https://user-images.githubusercontent.com/98681/163291233-530214ce-8188-461a-a1bd-406cc116d313.png">

<img width="1792" alt="Screen Shot 2022-04-13 at 5 31 35 PM" src="https://user-images.githubusercontent.com/98681/163291243-4b5bbbb7-4070-41b7-afe5-3ce7b2f15dc1.png">

GeoSot commented:
@mdo nice catch on algolia v3. 
Can you please split it to another PR to check it in separate (It works but doesn't get results)?

mdo commented:
Moved DocSearch changes to #36176.

GeoSot committed:
Separate examples code, Add some selectors, fix stackblitz btn

GeoSot commented:
It is very huge to be sure, but to the point I could follow, I didn't find something dysfunctional. 
So "nice job man" 🚀
